InvenTree/InvenTree/tasks.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import re
+import json
+import requests
+import logging
+
+from datetime import datetime, timedelta
+
+from django.core.exceptions import AppRegistryNotReady
+from django.db.utils import OperationalError, ProgrammingError
+
+
+logger = logging.getLogger(__name__)
+
+
+def schedule_task(taskname, **kwargs):
+    """
+    Create a scheduled task.
+    If the task has already been scheduled, ignore!
+    """
+
+    # If unspecified, repeat indefinitely
+    repeats = kwargs.pop('repeats', -1)
+
+    try:
+        from django_q.models import Schedule
+    except (AppRegistryNotReady):
+        logger.warning("Could not start background tasks - App registry not ready")
+        return
+
+    try:
+        if Schedule.objects.filter(func=taskname).exists():
+            logger.info(f"Scheduled task '{taskname}' already exists. (Skipping)")
+        else:
+            logger.info(f"Creating scheduled task '{taskname}'")
+
+            Schedule.objects.create(
+                name=taskname,
+                func=taskname,
+                repeats=repeats,
+                **kwargs
+            )
+    except (OperationalError, ProgrammingError):
+        # Required if the DB is not ready yet
+        pass
+
+
+def heartbeat():
+    """
+    Simple task which runs at 5 minute intervals,
+    so we can determine that the background worker
+    is actually running.
+
+    (There is probably a less "hacky" way of achieving this)?
+    """
+
+    try:
+        from django_q.models import Success
+        logger.warning("Could not perform heartbeat task - App registry not ready")
+    except AppRegistryNotReady:
+        return
+
+    threshold = datetime.now() - timedelta(minutes=30)
+
+    # Delete heartbeat results more than half an hour old,
+    # otherwise they just create extra noise
+    heartbeats = Success.objects.filter(
+        func='InvenTree.tasks.heartbeat',
+        started__lte=threshold
+    )
+
+    heartbeats.delete()
+
+
+def delete_successful_tasks():
+    """
+    Delete successful task logs
+    which are more than a month old.
+    """
+
+    try:
+        from django_q.models import Success
+        logger.warning("Could not perform 'delete_successful_tasks' - App registry not ready")
```

Comment:
logger statement needs to be after the except.

Reply:
Thanks!